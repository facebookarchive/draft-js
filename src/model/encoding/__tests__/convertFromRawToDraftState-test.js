/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 * @format
 * @oncall draft_js
 */

'use strict';

import type {EntityRange} from 'EntityRange';
import type {InlineStyleRange} from 'InlineStyleRange';
import type {RawDraftContentBlock} from 'RawDraftContentBlock';
import type {RawDraftContentState} from 'RawDraftContentState';

const convertFromRawToDraftState = require('convertFromRawToDraftState');
const mockUUID = require('mockUUID');

jest.mock('generateRandomKey');

const toggleExperimentalTreeDataSupport = (enabled: boolean) => {
  jest.doMock('gkx', () => name => {
    return name === 'draft_tree_data_support' ? enabled : false;
  });
};

const assertDraftState = (rawState: RawDraftContentState) => {
  expect(
    convertFromRawToDraftState(rawState).getBlockMap().toJS(),
  ).toMatchSnapshot();
};

beforeEach(() => {
  jest.resetModules();
  jest.mock('uuid', () => mockUUID);
});

test('must map falsey block types to default value of unstyled', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        text: 'AAAA',
        depth: 0,
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'B',
        text: 'BBBB',
        type: null,
        depth: 0,
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'C',
        text: 'CCCC',
        type: undefined,
        depth: 0,
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
    ],
    entityMap: {},
  };

  // $FlowFixMe looks like the whole point of the test is to verify something prevented by flow? Let it be for now.
  assertDraftState(rawState);
});

test('must be able to convert from styled blocks and entities mapped raw state', () => {
  const rawState = {
    blocks: [
      {
        data: {},
        depth: 0,
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
        key: 'a',
        text: 'Alpha',
        type: 'unstyled',
      },
      {
        data: {},
        depth: 0,
        entityRanges: [
          {
            key: 0,
            length: 5,
            offset: 0,
          },
        ],
        inlineStyleRanges: [
          {
            length: 5,
            offset: 0,
            style: 'BOLD',
          },
        ],
        key: 'b',
        text: 'Bravo',
        type: 'unordered-list-item',
      },
      {
        data: {},
        depth: 0,
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: [
          {
            length: 7,
            offset: 0,
            style: 'ITALIC',
          },
        ],
        key: 'c',
        text: 'Charlie',
        type: 'blockquote',
      },
    ],
    entityMap: {
      '0': {
        data: {},
        mutability: 'IMMUTABLE',
        type: 'IMAGE',
      },
    },
  };

  assertDraftState(rawState);
});

test('must convert from raw tree draft to raw content state when experimentalTreeDataSupport is disabled', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        text: '',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
        type: 'unstyled',
        depth: 0,
        children: [
          {
            key: 'B',
            text: '',
            entityRanges: ([]: Array<EntityRange>),
            inlineStyleRanges: ([]: Array<InlineStyleRange>),
            type: 'unstyled',
            depth: 0,
            children: [
              {
                key: 'C',
                text: 'left block',
                entityRanges: ([]: Array<EntityRange>),
                inlineStyleRanges: ([]: Array<InlineStyleRange>),
                type: 'unstyled',
                depth: 0,
                children: ([]: Array<RawDraftContentBlock>),
              },
              {
                key: 'D',
                text: 'right block',
                entityRanges: ([]: Array<EntityRange>),
                inlineStyleRanges: ([]: Array<InlineStyleRange>),
                type: 'unstyled',
                depth: 0,
                children: ([]: Array<RawDraftContentBlock>),
              },
            ],
          },
          {
            key: 'E',
            type: 'header-one',
            text: 'This is a tree based document!',
            entityRanges: ([]: Array<EntityRange>),
            inlineStyleRanges: ([]: Array<InlineStyleRange>),
            depth: 0,
            children: ([]: Array<RawDraftContentBlock>),
          },
        ],
      },
    ],
    entityMap: {},
  };

  assertDraftState(rawState);
});

test('convert from raw tree draft content state', () => {
  toggleExperimentalTreeDataSupport(true);
  const rawState = {
    blocks: [
      {
        key: 'A',
        text: '',
        entityRanges: ([]: Array<EntityRange>),
        depth: 0,
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
        type: 'unstyled',
        children: [
          {
            key: 'B',
            text: '',
            entityRanges: ([]: Array<EntityRange>),
            depth: 0,
            inlineStyleRanges: ([]: Array<InlineStyleRange>),
            type: 'unstyled',
            children: [
              {
                key: 'C',
                text: 'left block',
                entityRanges: ([]: Array<EntityRange>),
                depth: 0,
                inlineStyleRanges: ([]: Array<InlineStyleRange>),
                type: 'unstyled',
                children: ([]: Array<RawDraftContentBlock>),
              },
              {
                key: 'D',
                text: 'right block',
                entityRanges: ([]: Array<EntityRange>),
                depth: 0,
                inlineStyleRanges: ([]: Array<InlineStyleRange>),
                type: 'unstyled',
                children: ([]: Array<RawDraftContentBlock>),
              },
            ],
          },
          {
            key: 'E',
            type: 'header-one',
            text: 'This is a tree based document!',
            entityRanges: ([]: Array<EntityRange>),
            depth: 0,
            inlineStyleRanges: ([]: Array<InlineStyleRange>),
            children: ([]: Array<RawDraftContentBlock>),
          },
        ],
      },
    ],
    entityMap: {},
  };

  assertDraftState(rawState);
});

test('must be able to convert from raw state to tree state when experimentalTreeDataSupport is enabled', () => {
  toggleExperimentalTreeDataSupport(true);
  const rawState = {
    blocks: [
      {
        key: 'A',
        text: 'AAAA',
        entityRanges: ([]: Array<EntityRange>),
        type: 'unstyled',
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
        depth: 0,
      },
      {
        key: 'B',
        text: 'BBBB',
        entityRanges: ([]: Array<EntityRange>),
        type: 'unstyled',
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
        depth: 0,
      },
      {
        key: 'C',
        text: 'CCCC',
        entityRanges: ([]: Array<EntityRange>),
        type: 'unstyled',
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
        depth: 0,
      },
    ],
    entityMap: {},
  };

  assertDraftState(rawState);
});

test('must be able to convert content blocks that have list with depth from raw state to tree state when experimentalTreeDataSupport is enabled', () => {
  toggleExperimentalTreeDataSupport(true);
  const rawState = {
    blocks: [
      {
        key: 'A',
        type: 'ordered-list-item',
        depth: 0,
        text: '',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'B',
        type: 'ordered-list-item',
        depth: 1,
        text: '',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'C',
        type: 'ordered-list-item',
        depth: 2,
        text: 'deeply nested list',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
    ],
    entityMap: {},
  };

  assertDraftState(rawState);
});

test('ignore empty children array', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        type: 'ordered-list-item',
        depth: 0,
        text: 'A',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'B',
        type: 'ordered-list-item',
        depth: 0,
        text: 'B',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'C',
        type: 'ordered-list-item',
        depth: 0,
        text: 'C',
        children: ([]: Array<RawDraftContentBlock>),
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
    ],
    entityMap: {},
  };

  assertDraftState(rawState);
});

test('ignore empty children array for tree conversion 1', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        type: 'ordered-list-item',
        depth: 0,
        text: 'A',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'B',
        type: 'ordered-list-item',
        depth: 0,
        text: 'B',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'C',
        type: 'ordered-list-item',
        depth: 0,
        text: 'C',
        children: ([]: Array<RawDraftContentBlock>),
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
    ],
    entityMap: {},
  };
  assertDraftState(rawState);
});

test('ignore empty children array for tree conversion 2', () => {
  toggleExperimentalTreeDataSupport(true);
  const rawState = {
    blocks: [
      {
        key: 'A',
        type: 'ordered-list-item',
        depth: 0,
        text: 'A',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'B',
        type: 'ordered-list-item',
        depth: 0,
        text: 'B',
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
      {
        key: 'C',
        type: 'ordered-list-item',
        depth: 0,
        text: 'C',
        children: ([]: Array<RawDraftContentBlock>),
        entityRanges: ([]: Array<EntityRange>),
        inlineStyleRanges: ([]: Array<InlineStyleRange>),
      },
    ],
    entityMap: {},
  };
  assertDraftState(rawState);
});
