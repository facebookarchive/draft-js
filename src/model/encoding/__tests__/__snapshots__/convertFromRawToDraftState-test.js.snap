// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert from raw tree draft content state 1`] = `
Object {
  "A": Object {
    "characterList": Array [],
    "children": Array [
      "B",
      "E",
    ],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": null,
    "parent": null,
    "prevSibling": null,
    "text": "",
    "type": "unstyled",
  },
  "B": Object {
    "characterList": Array [],
    "children": Array [
      "C",
      "D",
    ],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "E",
    "parent": "A",
    "prevSibling": null,
    "text": "",
    "type": "unstyled",
  },
  "C": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": "D",
    "parent": "B",
    "prevSibling": null,
    "text": "left block",
    "type": "unstyled",
  },
  "D": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "D",
    "nextSibling": null,
    "parent": "B",
    "prevSibling": "C",
    "text": "right block",
    "type": "unstyled",
  },
  "E": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "E",
    "nextSibling": null,
    "parent": "A",
    "prevSibling": "B",
    "text": "This is a tree based document!",
    "type": "header-one",
  },
}
`;

exports[`ignore empty children array 1`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "B",
    "parent": null,
    "prevSibling": null,
    "text": "A",
    "type": "ordered-list-item",
  },
  "B": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "C",
    "parent": null,
    "prevSibling": "A",
    "text": "B",
    "type": "ordered-list-item",
  },
  "C": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "B",
    "text": "C",
    "type": "ordered-list-item",
  },
}
`;

exports[`ignore empty children array for tree conversion 1 1`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "B",
    "parent": null,
    "prevSibling": null,
    "text": "A",
    "type": "ordered-list-item",
  },
  "B": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "C",
    "parent": null,
    "prevSibling": "A",
    "text": "B",
    "type": "ordered-list-item",
  },
  "C": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "B",
    "text": "C",
    "type": "ordered-list-item",
  },
}
`;

exports[`ignore empty children array for tree conversion 2 1`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "B",
    "parent": null,
    "prevSibling": null,
    "text": "A",
    "type": "ordered-list-item",
  },
  "B": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "C",
    "parent": null,
    "prevSibling": "A",
    "text": "B",
    "type": "ordered-list-item",
  },
  "C": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "B",
    "text": "C",
    "type": "ordered-list-item",
  },
}
`;

exports[`must be able to convert content blocks that have list with depth from raw state to tree state when experimentalTreeDataSupport is enabled 1`] = `
Object {
  "A": Object {
    "characterList": Array [],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "key1",
    "parent": null,
    "prevSibling": null,
    "text": "",
    "type": "ordered-list-item",
  },
  "B": Object {
    "characterList": Array [],
    "children": Array [],
    "data": Object {},
    "depth": 1,
    "key": "B",
    "nextSibling": "key2",
    "parent": "key1",
    "prevSibling": null,
    "text": "",
    "type": "ordered-list-item",
  },
  "C": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 2,
    "key": "C",
    "nextSibling": null,
    "parent": "key2",
    "prevSibling": null,
    "text": "deeply nested list",
    "type": "ordered-list-item",
  },
  "key1": Object {
    "characterList": Array [],
    "children": Array [
      "B",
      "key2",
    ],
    "data": Object {},
    "depth": 0,
    "key": "key1",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "A",
    "text": "",
    "type": "ordered-list-item",
  },
  "key2": Object {
    "characterList": Array [],
    "children": Array [
      "C",
    ],
    "data": Object {},
    "depth": 1,
    "key": "key2",
    "nextSibling": null,
    "parent": "key1",
    "prevSibling": "B",
    "text": "",
    "type": "ordered-list-item",
  },
}
`;

exports[`must be able to convert from raw state to tree state when experimentalTreeDataSupport is enabled 1`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "B",
    "parent": null,
    "prevSibling": null,
    "text": "AAAA",
    "type": "unstyled",
  },
  "B": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "C",
    "parent": null,
    "prevSibling": "A",
    "text": "BBBB",
    "type": "unstyled",
  },
  "C": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "B",
    "text": "CCCC",
    "type": "unstyled",
  },
}
`;

exports[`must be able to convert from styled blocks and entities mapped raw state 1`] = `
Object {
  "a": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "a",
    "text": "Alpha",
    "type": "unstyled",
  },
  "b": Object {
    "characterList": Array [
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
      Object {
        "entity": "2",
        "style": Array [
          "BOLD",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "b",
    "text": "Bravo",
    "type": "unordered-list-item",
  },
  "c": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
      Object {
        "entity": null,
        "style": Array [
          "ITALIC",
        ],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "c",
    "text": "Charlie",
    "type": "blockquote",
  },
}
`;

exports[`must convert from raw tree draft to raw content state when experimentalTreeDataSupport is disabled 1`] = `
Object {
  "A": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "text": "",
    "type": "unstyled",
  },
  "B": Object {
    "characterList": Array [],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "text": "",
    "type": "unstyled",
  },
  "C": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "text": "left block",
    "type": "unstyled",
  },
  "D": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "D",
    "text": "right block",
    "type": "unstyled",
  },
  "E": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "E",
    "text": "This is a tree based document!",
    "type": "header-one",
  },
}
`;

exports[`must map falsey block types to default value of unstyled 1`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "text": "AAAA",
    "type": "unstyled",
  },
  "B": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "text": "BBBB",
    "type": "unstyled",
  },
  "C": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "text": "CCCC",
    "type": "unstyled",
  },
}
`;
